attachment_preview/static/src/js/components/chatter/chatter.esm.js

```diff
@@ -0,0 +1,272 @@
+        attachment_title
+    ) {
+        var url = "";
+        if (attachment_url) {
+            if (attachment_url.slice(0, 21) === "/web/static/lib/pdfjs") {
+                url = (window.location.origin || "") + attachment_url;
+            } else {
+                url =
+                    (window.location.origin || "") +
+                    "/attachment_preview/static/lib/ViewerJS/index.html" +
+                    "?type=" +
+                    encodeURIComponent(attachment_extension) +
+                    "&zoom=automatic" +
+                    "&title=" +
+                    encodeURIComponent(attachment_title) +
+                    "#" +
+                    attachment_url.replace(window.location.origin, "");
+            }
+        } else {
+            url =
+                (window.location.origin || "") +
+                "/attachment_preview/static/lib/ViewerJS/index.html" +
+                "?type=" +
+                encodeURIComponent(attachment_extension) +
+                "&title=" +
+                encodeURIComponent(attachment_title) +
+                "&zoom=automatic" +
+                "#" +
+                "/web/content/" +
+                attachment_id +
+                "?model%3Dir.attachment";
+        }
+
+        return url;
+    }
+
+    function canPreview(extension) {
+        return (
+            $.inArray(extension, [
+                "odt",
+                "odp",
+                "ods",
+                "fodt",
+                "pdf",
+                "ott",
+                "fodp",
+                "otp",
+                "fods",
+                "ots",
+            ]) > -1
+        );
+    }
+
+    patch(
+        components.Chatter.prototype,
+        "attachment_preview/static/src/js/components/chatter/chatter.js",
+        {
+            /**
+             * @override
+             */
+            constructor(...args) {
+                this._super(...args);
+
+                this._showPreview = this._showPreview.bind(this);
+                this.canPreview = this.canPreview.bind(this);
+            },
+
+            /**
+             * @override
+             */
+            _update() {
+                // Var res = this._super.apply(this, arguments);
+                var self = this;
+                self._getPreviewableAttachments().then(function (atts) {
+                    self.previewableAttachments = atts;
+                    self._updatePreviewButtons(self.previewableAttachments);
+                    if (!self.attachmentPreviewWidget) {
+                        self.attachmentPreviewWidget = new AttachmentPreviewWidget(
+                            self
+                        );
+                        self.attachmentPreviewWidget.setAttachments(atts);
+                    }
+                    self.previewableAttachments = atts;
+                    // ChatterpreviewableAttachments = atts;
+                    self.attachmentPreviewWidget.setAttachments(atts);
```

Comment:
I had to remove line 101, otherwise, as I was getting the following error message (browser debug console):

viewer.js:3547 offsetParent is not set -- cannot scroll